tests/src/Kernel/OauthTokenFileStorageTest.php

```diff
@@ -0,0 +1,235 @@
+    $this->assertSame($this->testTokenData['refresh_token'], $stored_token['refresh_token']);
+    $this->assertSame($this->testTokenData['scope'], $stored_token['scope']);
+    // The difference in the timestamp should be 1 or 0 seconds.
+    $this->assertLessThan(2, abs($this->testTokenData['expires_in'] + $current_time - $stored_token['expires']));
+  }
+
+  /**
+   * Test the token storage is using static cache.
+   */
+  public function testStaticCaching() {
+    $storage = $this->tokenStorage();
+    // Save the token.
+    $storage->saveToken($this->testTokenData);
+
+    $access_token = $storage->getAccessToken();
+
+    // Load raw token data.
+    $stored_token = unserialize(base64_decode(file_get_contents($this->tokenFileUri())));
+
+    // Create a new access token and write it to file.
+    $stored_token['access_token'] = strtolower($this->randomMachineName(32));
+    file_unmanaged_save_data(base64_encode(serialize($stored_token)), $this->tokenFileUri(), FILE_EXISTS_REPLACE);
+
+    // Make sure the cached version is still returned.
+    $this->assertSame($access_token, $storage->getAccessToken());
+  }
+
+  /**
+   * Test the get and has methods of the cache storage.
+   */
+  public function testGetters() {
+    $storage = $this->tokenStorage();
+    // Will use this to test expire.
+    $current_time = time();
+    // Save the token.
+    $storage->saveToken($this->testTokenData);
+
+    $this->assertSame($this->testTokenData['access_token'], $storage->getAccessToken());
+    $this->assertSame($this->testTokenData['token_type'], $storage->getTokenType());
+    $this->assertSame($this->testTokenData['refresh_token'], $storage->getRefreshToken());
+    $this->assertSame($this->testTokenData['scope'], $storage->getScope());
+    // The difference in the timestamp should be 1 or 0 seconds.
+    $this->assertLessThan(2, abs($this->testTokenData['expires_in'] + $current_time - $storage->getExpires()));
+
+    // The token should still be valid for 5 minutes.
+    $this->assertFalse($storage->hasExpired());
+
+    // Expire token.
+    $storage->markExpired();
+
+    // The token should still be valid for 5 minutes.
+    $this->assertTrue($storage->hasExpired());
+  }
+
+  /**
+   * Tests OAuth token data file removal on module uninstall.
+   */
+  public function testTokenFileRemovalOnUninstall() {
+    $storage = $this->tokenStorage();
+    // Save the token.
+    $storage->saveToken($this->testTokenData);
+    $this->assertTrue(file_exists($this->tokenFileUri()), 'Oauth token data file should not exist after the module got uninstalled.');
+    /** @var \Drupal\Core\Extension\ModuleInstallerInterface $installer */
+    $installer = $this->container->get('module_installer');
+    $installer->uninstall(['apigee_edge']);
+    $this->assertFalse(file_exists($this->tokenFileUri()), 'Oauth token data file should not exist after the module got uninstalled.');
+  }
+
```

Comment:
Is there a good reason the `testCacheClear` test was removed? There should still be coverage for that.

Reply:
See the related  TODO in the top of the file: https://github.com/apigee/apigee-edge-drupal/pull/94/files#diff-0a036b86ad28ed8eb12935d8098980e0R33